src/devices/src/legacy/rtc.rs

```diff
@@ -0,0 +1,32 @@
+use std::convert::TryInto;
+use vm_device::bus::MmioAddress;
+use vm_device::MutDeviceMmio;
+use vm_superio::{rtc_pl031::NoEvents, RTC};
+
+use utils::debug;
+
+pub struct RTCWrapper(pub RTC<NoEvents>);
+
+impl MutDeviceMmio for RTCWrapper {
+    fn mmio_read(&mut self, _base: MmioAddress, offset: u64, data: &mut [u8]) {
+        if data.len() != 4 {
+            debug!("RTC invalid data length on read: {}", data.len());
+        }
+
+        match offset.try_into() {
+            Ok(offset) => self.0.read(offset, data.try_into().unwrap()),
```

Comment:
Can we leave a comment explaining why is the unwrap safe?

Reply:
Added a comment.